### ISSUE TITLE
Race condition with NRF_802154_CSMA_CA_WAIT_FOR_TIMESLOT config option enabled

### ISSUE THREAD

**Issue by hubertmis** [NordicSemiconductor/nRF-IEEE-802.15.4-radio-driver#197]
During stress tests we've hit a race condition with NRF_802154_CSMA_CA_WAIT_FOR_TIMESLOT configuration option enabled when used in multiprotocol configuration. Suggested workaround for the race condition is to increase safety margin of the timeslot (`timeslot_safe_margin`):
```
const nrf_raal_softdevice_cfg_t cfg = {
.timeslot_length = NRF_RAAL_TIMESLOT_DEFAULT_LENGTH,
.timeslot_timeout = NRF_RAAL_TIMESLOT_DEFAULT_TIMEOUT,
.timeslot_max_length = NRF_RAAL_TIMESLOT_DEFAULT_MAX_LENGTH,
.timeslot_alloc_iters = NRF_RAAL_TIMESLOT_DEFAULT_ALLOC_ITERS,
.timeslot_safe_margin = nrf_raal_softdevice_safe_margin_calc(NRF_RAAL_DEFAULT_LF_CLK_ACCURACY_PPM) + 31,
.lf_clk_accuracy_ppm = NRF_RAAL_DEFAULT_LF_CLK_ACCURACY_PPM,
};
nrf_raal_softdevice_config(&cfg);

**Comment by hubertmis** [comment]
I hope this problem could be fixed by #214 . Should be verified after merge.

**Comment by hubertmis** [comment]
According to current tests, this problem seems fixed.
